modules/crud/sql/update.go

```diff
@@ -1,60 +1,226 @@
+//generateUpdateQuery makes query for update operations
+func (s *SQL) generateUpdateQuery(ctx context.Context, project, col string, req *model.UpdateRequest, op string) (string, []interface{}, error) {
 	// Generate a prepared query builder
-	query := goqu.From(col).Prepared(true)
-	query = query.SetAdapter(goqu.NewAdapter(s.dbType, query))
+	dialect := goqu.Dialect(s.dbType)
+	query := dialect.From(col)
+	if op == "$set" {
+		query = query.Prepared(true)
+	}
 
 	if req.Find != nil {
-
 		// Get the where clause from query object
 		var err error
 		query, err = generateWhereClause(query, req.Find)
 		if err != nil {
 			return "", nil, err
 		}
-
 	}
 
 	if req.Update == nil {
 		return "", nil, utils.ErrInvalidParams
 	}
+	m, ok := req.Update[op].(map[string]interface{})
+	if !ok {
+		return "", nil, utils.ErrInvalidParams
+	}
 
-	record, err := generateRecord(req.Update["$set"])
+	if op == "$currentDate" {
+		for k,v := range m {
+			mm, ok := v.(map[string]interface{})
+			if !ok {
+				return "", nil, utils.ErrInvalidParams
+			}
+			for _, val := range mm {
+				val, ok := val.(string)
+				if !ok {
+					return "", nil, utils.ErrInvalidParams
+				}
+				switch val {
+				case "date", "timestamp":
+					m[k] = val
+				default:
+					return "", nil, utils.ErrInvalidParams
+				}
+			}
+		}
+	}
+	
+	record, err := generateRecord(req.Update[op])
 	if err != nil {
 		return "", nil, err
 	}
 
 	// Generate SQL string and arguments
-	sqlString, args, err := query.ToUpdateSql(record)
+	sqlString, args, err := query.Update().Set(record).ToSQL()
 	if err != nil {
 		return "", nil, err
 	}
 
 	sqlString = strings.Replace(sqlString, "\"", "", -1)
+	switch op {
+	case "$set":
+	case "$inc":
+		for k, v := range m {
+			val, err := numToString(v)
+			if err != nil {
+				return "", nil, err
+			}
+			sqlString = strings.Replace(sqlString, k+"="+val, k+"="+k+"+"+val, -1)
+		}
+	case "$mul":
+		for k, v := range m {
+			val, err := numToString(v)
+			if err != nil {
+				return "", nil, err
+			}
+			sqlString = strings.Replace(sqlString, k+"="+val, k+"="+k+"*"+val, -1)
+		}
+	case "$max":
+		for k, v := range m {
+			val, err := numToString(v)
+			if err != nil {
+				return "", nil, err
+			}
+			sqlString = strings.Replace(sqlString, k+"="+val, k+"=GREATEST("+k+","+val+")", -1)
+		}
+	case "$min":
+		for k, v := range m {
+			val, err := numToString(v)
+			if err != nil {
+				return "", nil, err
+			}
+			sqlString = strings.Replace(sqlString, k+"="+val, k+"=LEAST("+k+","+val+")", -1)
+		}
+	case "$currentDate":
+		for k, v := range m {
+			val, ok := v.(string)
+			if !ok {
+				return "", nil, utils.ErrInvalidParams
+			}
+			if val == "timestamp" {
+				sqlString = strings.Replace(sqlString, k+"='"+val+"'", k+"=CURRENT_TIMESTAMP()", -1)
+			} else {
+				sqlString = strings.Replace(sqlString, k+"='"+val+"'", k+"=CURRENT_DATE()", -1)
```

Comment:
Do we have `CURRENT_DATE()` and `CURRENT_TIMESTAMP()` exist for both, postgres and MySQL?